### ISSUE TITLE
Error disassembling file

### ISSUE THREAD

**Issue by lab313ru** [rmtew/peasauce#10]
```
003828 (003828): Found end of block boundary with processed code and no end instruction (data type: 5, processed: 16)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1973, in new_thread_wrapper
    func(*posargs, **kwargs)
  File "C:\Python27\lib\threading.py", line 774, in __bootstrap
    self.__bootstrap_inner()
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "d:\git\peasauce\python\disassembly_util.py", line 68, in run
    traceback.print_stack()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 888, in trace_func
    return self._events[event](frame, arg)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1067, in handle_line
    self.block_maybe_attach()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1168, in block_maybe_attach
    self.block(block_cond)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1215, in block
    self.unblock_work()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1241, in <lambda>
    self.schedule_work(lambda : self.run_locally(text, cur_frame, execution_id, frame_kind, repr_kind))
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1300, in run_locally
    res = eval(code, cur_frame.f_globals, self.get_locals(cur_frame, frame_kind))
  File "<debug input>", line 1, in <module>
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1973, in new_thread_wrapper
    func(*posargs, **kwargs)
  File "C:\Python27\lib\threading.py", line 774, in __bootstrap
    self.__bootstrap_inner()
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "d:\git\peasauce\python\disassembly_util.py", line 68, in run
    traceback.print_stack()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 888, in trace_func
    return self._events[event](frame, arg)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1067, in handle_line
    self.block_maybe_attach()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1168, in block_maybe_attach
    self.block(block_cond)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1215, in block
    self.unblock_work()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1241, in <lambda>
    self.schedule_work(lambda : self.run_locally(text, cur_frame, execution_id, frame_kind, repr_kind))
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1300, in run_locally
    res = eval(code, cur_frame.f_globals, self.get_locals(cur_frame, frame_kind))
  File "<debug input>", line 1, in <module>
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1973, in new_thread_wrapper
    func(*posargs, **kwargs)
  File "C:\Python27\lib\threading.py", line 774, in __bootstrap
    self.__bootstrap_inner()
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "d:\git\peasauce\python\disassembly_util.py", line 68, in run
    traceback.print_stack()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 888, in trace_func
    return self._events[event](frame, arg)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1067, in handle_line
    self.block_maybe_attach()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1168, in block_maybe_attach
    self.block(block_cond)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1215, in block
    self.unblock_work()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1241, in <lambda>
    self.schedule_work(lambda : self.run_locally(text, cur_frame, execution_id, frame_kind, repr_kind))
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1300, in run_locally
    res = eval(code, cur_frame.f_globals, self.get_locals(cur_frame, frame_kind))
  File "<debug input>", line 1, in <module>
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1973, in new_thread_wrapper
    func(*posargs, **kwargs)
  File "C:\Python27\lib\threading.py", line 774, in __bootstrap
    self.__bootstrap_inner()
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "d:\git\peasauce\python\disassembly_util.py", line 68, in run
    traceback.print_stack()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 888, in trace_func
    return self._events[event](frame, arg)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1067, in handle_line
    self.block_maybe_attach()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1168, in block_maybe_attach
    self.block(block_cond)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1215, in block
    self.unblock_work()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1241, in <lambda>
    self.schedule_work(lambda : self.run_locally(text, cur_frame, execution_id, frame_kind, repr_kind))
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1300, in run_locally
    res = eval(code, cur_frame.f_globals, self.get_locals(cur_frame, frame_kind))
  File "<debug input>", line 1, in <module>
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1973, in new_thread_wrapper
    func(*posargs, **kwargs)
  File "C:\Python27\lib\threading.py", line 774, in __bootstrap
    self.__bootstrap_inner()
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "d:\git\peasauce\python\disassembly_util.py", line 68, in run
    traceback.print_stack()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 888, in trace_func
    return self._events[event](frame, arg)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1067, in handle_line
    self.block_maybe_attach()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1168, in block_maybe_attach
    self.block(block_cond)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1215, in block
    self.unblock_work()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1241, in <lambda>
    self.schedule_work(lambda : self.run_locally(text, cur_frame, execution_id, frame_kind, repr_kind))
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1300, in run_locally
    res = eval(code, cur_frame.f_globals, self.get_locals(cur_frame, frame_kind))
  File "<debug input>", line 1, in <module>
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1973, in new_thread_wrapper
    func(*posargs, **kwargs)
  File "C:\Python27\lib\threading.py", line 774, in __bootstrap
    self.__bootstrap_inner()
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "d:\git\peasauce\python\disassembly_util.py", line 68, in run
    traceback.print_stack()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1973, in new_thread_wrapper
    func(*posargs, **kwargs)
  File "C:\Python27\lib\threading.py", line 774, in __bootstrap
    self.__bootstrap_inner()
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "d:\git\peasauce\python\disassembly_util.py", line 69, in run
    raise
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 888, in trace_func
    return self._events[event](frame, arg)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1067, in handle_line
    self.block_maybe_attach()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1168, in block_maybe_attach
    self.block(block_cond)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1215, in block
    self.unblock_work()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1241, in <lambda>
    self.schedule_work(lambda : self.run_locally(text, cur_frame, execution_id, frame_kind, repr_kind))
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1300, in run_locally
    res = eval(code, cur_frame.f_globals, self.get_locals(cur_frame, frame_kind))
  File "<debug input>", line 1, in <module>
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1973, in new_thread_wrapper
    func(*posargs, **kwargs)
  File "C:\Python27\lib\threading.py", line 774, in __bootstrap
    self.__bootstrap_inner()
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "d:\git\peasauce\python\disassembly_util.py", line 70, in run
    except SystemExit:
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 888, in trace_func
    return self._events[event](frame, arg)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1067, in handle_line
    self.block_maybe_attach()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1168, in block_maybe_attach
    self.block(block_cond)
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1215, in block
    self.unblock_work()
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1241, in <lambda>
    self.schedule_work(lambda : self.run_locally(text, cur_frame, execution_id, frame_kind, repr_kind))
  File "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Python Tools for Visual Studio\2.2\visualstudio_py_debugger.py", line 1300, in run_locally
    res = eval(code, cur_frame.f_globals, self.get_locals(cur_frame, frame_kind))
  File "<debug input>", line 1, in <module>
Exception in thread Thread-1:
Traceback (most recent call last):
  File "C:\Python27\lib\threading.py", line 801, in __bootstrap_inner
    self.run()
  File "d:\git\peasauce\python\disassembly_util.py", line 65, in run
    completed_event.result = work_data[0](*work_data[1], **work_data[2])
  File "d:\git\peasauce\python\disassembly.py", line 1799, in api_load_file
    platform_specific_processing(program_data)
  File "d:\git\peasauce\python\disassembly.py", line 1816, in platform_specific_processing
    platform_specific_processing_M680x0_amiga(program_data, work_state)
  File "d:\git\peasauce\python\disassembly.py", line 1899, in platform_specific_processing_M680x0_amiga
    address_register_values[current_dest_register_number] = current_source_operand_values["xxx"][0]
KeyError: 'xxx'
```

[ppami4.zip](https://github.com/rmtew/peasauce/files/768988/ppami4.zip)

**Comment by rmtew** [comment]
You need to provide reproduction steps.  Just a zip and a stacktrace tells me nothing about how to make the problem happen.

**Comment by lab313ru** [comment]
Just Open file and, after some time it gives that error, and stucks at 90%.

**Comment by rmtew** [comment]
Thanks for the report.

A recent feature is that after doing the raw disassembling, it will try and process the code and automatically rename library names to things like "DosLibName" or "GraphicsLibName", and flags what addresses are used to store the library handles.  With a bit of extra code that is still to be written, it would use fd files to track what calls are made with the handles, and rename the offsets - and some more stuff.

However, it is quite simple at this time.  It doesn't follow too far back to find what library address or name is being used, and gives up.  And it doesn't know what to do with C compiled code where library handles (including execbase in this case) are stored in the A4 memory block.

It also errored on weird C use of assembly, like in this case LEA D16(PC), AR.  It will no longer error on this instruction, but it still won't produce any useful library call identification results, due to the A4 memory stuff.

So, TLDR; it should no longer error.  Thanks for the bug report.
